### PR TITLE
Add support for Delinea Secret Server

### PR DESCRIPTION
**Pull request by pacificcode**
## Problem Statement
What is the problem you're trying to solve?

This PR addresses https://github.com/external-secrets/external-secrets/issues/1709
Adding support for Delinea Secret Server

e2e tests run successfully, however a fully functional Secret Server account is necessary to run them.
To acquire a username and password, refer to the Secret Server [user management](https://docs.delinea.com/online-help/secret-server/users/creating-users/index.htm) documentation.

There is also useful information here to create a [trial account](https://delinea.com/buy-try?&qgad=681248112387&qgterm=secret%20server&utm_medium=Paid-Search&utm_source=Google&utm_campaign=NAM_Brand&utm_content=Secret-Server_PM_KW&utm_term=secret%20server%7Cp%7Cg%7Cc%7C%7C681248112387&hsa_cam=16000616630&hsa_grp=154989480416&hsa_mt=p&hsa_src=g&hsa_ad=681248112387&hsa_acc=%7B4968234215%7D&hsa_net=adwords&hsa_kw=secret%20server&hsa_tgt=kwd-23934096104&hsa_ver=3&gad_source=1&gclid=EAIaIQobChMIs57m2Z_8hQMVVs3CBB3xkAHgEAAYASABEgJJFPD_BwE&gclsrc=aw.ds)

## Checklist

- [ √] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [ √] All commits are signed with `git commit --signoff`
- [ √] My changes have reasonable test coverage
- [ √] All tests pass with `make test`
- [ √] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

**Comment by gusfcarvalho** [comment]
@pacificcode ideally we want e2e tests to run everytime on a new PR - is it possible to get a permanent account in order to run them? 😅 If not, we will double check on costs to see if it's viable to pay for them with the donation amounts that we receive monthly.

**Comment by pacificcode** [comment]
> @pacificcode ideally we want e2e tests to run everytime on a new PR - is it possible to get a permanent account in order to run them? 😅 If not, we will double check on costs to see if it's viable to pay for them with the donation amounts that we receive monthly.

@gusfcarvalho I passed on your inquiry to management. Someone will get back to you when information is available.

**Comment by gusfcarvalho** [comment]
> @gusfcarvalho I passed on your inquiry to management. Someone will get back to you when information is available.

Please DM me in kubernetes slack (@gusfcarvalho) if you need anything specific. One way this could be arranged is a donation to our Open Collective org to an amount similar to the subscription (I think that works for us - C.C. @moolen )

**Comment by pacificcode** [comment]
@gusfcarvalho @moolen is there anything we can do to get this PR reviewed? Im happy to help in anyway requested.

**Comment by gusfcarvalho** [comment]
@pacificcode sorry for taking so long. Configured env vars - let's see if e2e tests run smoothly 😄

File: Tiltfile [hunk 1]
```diff
@@ -9,7 +9,7 @@ if str(local("command -v " + kubectl_cmd + " || true", quiet = True)) == "":
 # set defaults
 settings = {
     "debug": {
-        "enabled": False,
+        "enabled": True,
```
Comment (Skarlso) [inline Tiltfile:12]: You can use a local tilt setting file for this instead of modifying the actual tilt file. :)

Comment (pacificcode) [reply Tiltfile]: @Skarlso good catch, this is resolved.

File: Tiltfile [hunk 2]
```diff
@@ -51,7 +51,7 @@ if settings.get('debug').get('enabled'):
 
 local_resource(
     'external-secret-binary',
-    "CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -gcflags '{gcflags}' -v -o bin/external-secrets ./".format(gcflags=gcflags),
```
Comment (Skarlso) [inline Tiltfile:54]: Also, please remove this modification. Alternatively, make it a setting in tilt setting file.

**Comment by jcralbino** [comment]
Hello
What is the status of this integration?
I am a user of Delinea secret server and I will like to use the oci images

**Comment by gusfcarvalho** [comment]
e2e run https://github.com/external-secrets/external-secrets/actions/runs/9700910151/job/26773388382

File: hack/api-docs/mkdocs.yml [hunk 3]
```diff
@@ -116,6 +116,10 @@ nav:
       - Cloak End 2 End Encrypted Secrets: provider/cloak.md
       - Scaleway: provider/scaleway.md
       - Delinea: provider/delinea.md
+<<<<<<< HEAD
+      - Secret Server: provider/delinea.md
+=======
+>>>>>>> main
```
Comment (gusfcarvalho) [inline hack/api-docs/mkdocs.yml:122]:
```suggestion
      - Secret Server: provider/delinea.md
```

**Comment by gusfcarvalho** [comment]
[Bot] - :x: [e2e tests failed](https://github.com/external-secrets/external-secrets/actions/runs/)

**Comment by gusfcarvalho** [comment]
[Bot] - :x: [e2e tests failed](https://github.com/external-secrets/external-secrets/actions/runs/)

**Comment by gusfcarvalho** [comment]
[Bot] - :x: [e2e tests failed](https://github.com/external-secrets/external-secrets/actions/runs/)

**Comment by gusfcarvalho** [comment]
[Bot] - :x: [e2e tests failed](https://github.com/external-secrets/external-secrets/actions/runs/)

**Comment by gusfcarvalho** [comment]
[Bot] - :x: [e2e tests failed](https://github.com/external-secrets/external-secrets/actions/runs/)

**Comment by gusfcarvalho** [comment]
[Bot] - :white_check_mark: [e2e tests pass](https://github.com/external-secrets/external-secrets/actions/runs/)

**Comment by gusfcarvalho** [comment]
[Bot] - :x: [e2e tests failed](https://github.com/external-secrets/external-secrets/actions/runs/)

**Review by gusfcarvalho — APPROVED** [review]

**Comment by awickham10** [comment]
@pacificcode the documentation on this references Secret Server cloud. Do you know if this will work with a self-hosted Secret Server installation?

**Comment by srekkas** [comment]
@pacificcode how to use with on premises SecretServer?

**Comment by awickham10** [comment]
> @awickham10 Just use yours on premise SecretServer url and it works

That worked for us. Thank you.

**Pull request merged**